access-grant/src/main/java/com/inrupt/client/accessgrant/Utils.java

```diff
@@ -119,6 +133,41 @@ public static List<Map<String, Object>> getCredentialsFromPresentation(final Map
         return credentials;
     }
 
+    public static int convertPage(final String page) {
+        try {
+            final int p = Integer.parseInt(page);
```

Comment:
I thought that page was going to be an opaque values and not something a client should parse. Does the client need to do this or should it follow the links only.

Reply:
This is part of the fallback mechanism for cases where a client is using the client paging API over a non-page-able server API (i.e. /derive). Here, in order to satisfy the expectations of the new response object, the client will do some internal "paging". This could certainly be an opaque value, though it's also not "true" paging.

It's worth noting that the response value here is not a URI, but rather a `CredentialFilter<T>` that can be passed to another invocation of `::query`

Reply:
One option would be to create a random internal value as part of a client instance and use that as part of the basis for paging. But that also seemed like over-complicating a fall-back mechanism.

Reply:
This code has been removed via https://github.com/inrupt/solid-client-java/pull/1684/commits/9a97e52b0910518bb41d3a412e2516c25a0d507a